### DISCUSSION TITLE
New System Install and Init

### DISCUSSION THREAD

**Discussion by jazzledark** [ni/nidaqmx-python#638] [Q&A] [answered]
Interesting observation using this library on a fresh Ubuntu machine. 

1. Install and update Ubuntu 22
2. Install 2024Q3 Stream repos from NI
3. Install nidaqmx using apt - You get these kernel modules loaded but other tools like nisyscfg will not function
<img width="775" alt="image" src="https://github.com/user-attachments/assets/9dbf96ef-ad42-447a-a4e2-7a2df3b3bb51">

4. If you then install nidaqmx-python and run something like add_network_device it seems the rest of the kernel modules get loaded the system gets "initialized"

<img width="1405" alt="image" src="https://github.com/user-attachments/assets/6082b61a-8b02-48b1-bad4-af99b48465a8">

Would be really nice for new setups if I could init the local nidaqmx system without having to call add_network_device. Is this possible? Is this the expected behavior?

**Comment by zhindes** [comment, marked as answer]
Do you have hardware plugged into your system (e.g., an NI PCI or PXI device)? Or are you only using network based DAQ devices (e.g., Network cDAQ or FieldDAQ)? If the latter, then this is expected. Until you Add (and Reserve) the device, there is no software loaded.

**Reply by jazzledark** [reply]
Only network cDAQ - so it seems to be as designed. My understanding is that there is no way to init the system without a direct attached device or add/reserve. No init function exists. Appreciate your prompt response above. Cheers.
